modules/csvparser/filterx-func-parse-csv.c

```diff
@@ -0,0 +1,346 @@
+{
+  FilterXFunctionParseCSV *self = (FilterXFunctionParseCSV *) s;
+  filterx_expr_unref(self->msg);
+  filterx_expr_unref(self->columns);
+  csv_scanner_options_clean(&self->options);
+  filterx_function_free_method(&self->super);
+}
+
+static FilterXExpr *
+_extract_msg_expr(FilterXFunctionArgs *args, GError **error)
+{
+  FilterXExpr *msg_expr = filterx_function_args_get_expr(args, 0);
+  if (!msg_expr)
+    {
+      g_set_error(error, FILTERX_FUNCTION_ERROR, FILTERX_FUNCTION_ERROR_CTOR_FAIL,
+                  "argument must be set: msg_str. " FILTERX_FUNC_PARSE_CSV_USAGE);
+      return NULL;
+    }
+
+  return msg_expr;
+}
+
+static FilterXExpr *
+_extract_columns_expr(FilterXFunctionArgs *args, GError **error)
+{
+  return filterx_function_args_get_named_expr(args, FILTERX_FUNC_PARSE_CSV_ARG_NAME_COLUMNS);
+}
+
+static gboolean
+_extract_opts(FilterXFunctionParseCSV *self, FilterXFunctionArgs *args, GError **error)
+{
+  guint32 opt_flags = self->options.flags;
+
+  const gchar *error_str = "";
+  gboolean exists;
+  gsize len;
+  const gchar *value;
+  FilterXObject *obj;
+
+  value = filterx_function_args_get_named_literal_string(args, FILTERX_FUNC_PARSE_CSV_ARG_NAME_DELIMITERS, &len, &exists);
+  if (exists)
+    {
+      if (len < 1)
+        {
+          error_str = FILTERX_FUNC_PARSE_CSV_ARG_NAME_DELIMITERS " can not be empty";
+          goto error;
+        }
+      if (!value)
+        {
+          error_str = FILTERX_FUNC_PARSE_CSV_ARG_NAME_DELIMITERS " must be a string literal";
+          goto error;
+        }
+      csv_scanner_options_set_delimiters(&self->options, value);
+    }
+
+  value = filterx_function_args_get_named_literal_string(args, FILTERX_FUNC_PARSE_CSV_ARG_NAME_DIALECT, &len, &exists);
+  if (exists)
+    {
+      if (len < 1)
+        {
+          error_str = FILTERX_FUNC_PARSE_CSV_ARG_NAME_DIALECT " can not be empty";
+          goto error;
+        }
+      if (!value)
+        {
+          error_str = FILTERX_FUNC_PARSE_CSV_ARG_NAME_DIALECT " must be a string literal";
+          goto error;
+        }
+      CSVScannerDialect dialect = csv_parser_lookup_dialect(value);
+      if (dialect == -1)
+        {
+          error_str = FILTERX_FUNC_PARSE_CSV_ARG_NAME_DIALECT " argument must be one of: [" \
+                      "escape-none, " \
+                      "escape-backslash, " \
+                      "escape-backslash-with-sequences, " \
+                      "escape-double-char]";
+          goto error;
+        }
+      csv_scanner_options_set_dialect(&self->options, dialect);
+    }
+
+  obj = filterx_function_args_get_named_object(args, FILTERX_FUNC_PARSE_CSV_ARG_NAME_GREEDY, &exists);
+  if (exists)
+    {
+      if (!obj)
+        {
+          error_str = FILTERX_FUNC_PARSE_CSV_ARG_NAME_GREEDY " argument evaluation error";
+          goto error;
+        }
+      if (filterx_object_truthy(obj))
+        opt_flags |= CSV_SCANNER_GREEDY;
+      else
+        opt_flags &= ~CSV_SCANNER_GREEDY;
```

Comment:
I would use `filterx_function_args_get_named_literal_boolean()` here, to make sure it is a boolean object. Same for `strip_whitespaces`.